### PR TITLE
Add driver.yml - the workflow to install eBPF core on a self-host machine.

### PR DESCRIPTION
**Pull request by Alan-Jowett**
Add driver.yml - the workflow to install eBPF core on a self-host machine.

Signed-off-by: Alan Jowett <alanjo@microsoft.com>

### CODE REVIEW

**Comment by Alan-Jowett** [comment]
This PR partially addresses #141.

It loads the ebpfcore.sys and netebpfext.sys.

**Comment by Alan-Jowett** [comment]
> This PR partially addresses #141.
> 
> It loads the ebpfcore.sys and netebpfext.sys.

Added call to existing "run_tests.bat" script that executes all the driver CI/CD tests.

File: docs/SelfHostedRunnerSetup.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Setup instructions for self-hosted runners
+
+Self-hosted runners are necessary as GitHub hosted runners don't have the requisite permissions to install drivers.
+
+1) Install Windows Server 2019 - build 17763.
```
Comment (shankarseal) [inline docs/SelfHostedRunnerSetup.md:5]: link to a vhdx maybe? Or some install point?

File: .github/workflows/driver_test.yml [hunk 2]
```diff
@@ -20,14 +20,26 @@ jobs:
     strategy:
       matrix:
         configurations: [Debug, Release]
-    runs-on: windows-latest
+
+    runs-on: [windows, self-hosted]
```
Comment (dthaler) [inline .github/workflows/driver_test.yml:24]: Line 4 points to information on the github environment.  Does that apply to self-hosted?  If not, replace that comment with one that provides information on how self-hosted works.   Maybe point to your SelfHostedRunnerSetup.md doc?

File: docs/SelfHostedRunnerSetup.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+2) Enable Test Signing mode.
+   1) ``` bcdedit /set testsigning on ```
+3) Enable driver verifier.
+   1) ``` verifier /standard /bootmode persistent /driver ebpfcore.sys netebpfext.sys  ```
```
Comment (shankarseal) [inline docs/SelfHostedRunnerSetup.md:9]: and sample_ebpf_ext.sys

File: docs/SelfHostedRunnerSetup.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+   5) ```Add-Type -AssemblyName System.IO.Compression.FileSystem ; [System.IO.Compression.ZipFile]::ExtractToDirectory("$PWD/actions-runner-win-x64-2.281.1.zip", "$PWD")```
+6) Obtain [authentication token](https://github.com/microsoft/ebpf-for-windows/settings/actions/runners/new). This requires administrator permissions in the project.
+7) Configure action runner.
+   1) ```./config.cmd --url https://github.com/microsoft/ebpf-for-windows --token <action runner token>```
+8) Change action runner service to run as "LocalSystem".
+   1) Open services.msc.
+   2) Locate "GitHub Action Runner ...".
+   3) Right Click -> Properties -> Log On.
+   4) Change "Log on as:" to "Local System Account".
+9) Reboot the runner.
```
Comment (Alan-Jowett) [inline docs/SelfHostedRunnerSetup.md:25]: Add step to install the Visual C++ redistributable runtime.

File: .github/workflows/driver_test.yml [hunk 5]
```diff
@@ -20,14 +20,26 @@ jobs:
...
     steps:
-    - name: Download build output
-      uses: actions/download-artifact@v2
+    - name: Download workflow artifact
+      uses: dawidd6/action-download-artifact@v2.14.1
       with:
+        workflow: build.yml
+        workflow_conclusion: success
```
[the remark this answers is not in the document]
Comment (Alan-Jowett) [reply .github/workflows/driver_test.yml:31]: Added success condition.

**Comment by Alan-Jowett** [comment]
Test results:
https://github.com/Alan-Jowett/ebpf-for-windows/actions/runs/1218694988

**Comment by Alan-Jowett** [comment]
Updated test results, with tests passing:
https://github.com/Alan-Jowett/ebpf-for-windows/runs/3561998206

**Review by dthaler — APPROVED** [review]

**Review by shankarseal — APPROVED** [review]

**Pull request merged**